File: LeetCode/cpp/1637_Widest_Vertical_Area_Between_Two_Points_Containing_No_Points.cpp
class Solution {
public:
	int maxWidthOfVerticalArea(vector<vector<int>>& points) {
		int answer = 0;
		map<int, int> mp;

		for (vector<int> point : points) {
			mp[point[0]]++;
		}

		int prev = -1;

		for (pair<int, int> p : mp) {
			if (prev == -1) {
				prev = p.first;
				continue;
			}

			answer = max(answer, p.first - prev);
			prev = p.first;
		}

		return answer;
	}
};

File: LeetCode/java/1637_Widest_Vertical_Area_Between_Two_Points_Containing_No_Points.java
import java.util.Map;
import java.util.TreeMap;

class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        int answer = 0;
        Map<Integer, Integer> map = new TreeMap<>();

        for (int[] point : points) {
            map.put(point[0], map.getOrDefault(point[0], 0) + 1);
        }

        int prev = -1;

        for (int key : map.keySet()) {
            if (prev == -1) {
                prev = key;
                continue;
            }

            answer = Math.max(answer, key - prev);
            prev = key;
        }

        return answer;
    }
}